autotest/ogr/ogr_mssqlspatial.py

```diff
@@ -600,6 +600,52 @@ def test_ogr_mssqlspatial_bulk_insert():
 #
 
 
+def test_ogr_mssqlspatial_geography_polygon_vertex_order():
+    """Test issue GH https://github.com/OSGeo/gdal/issues/1128"""
+
+    if gdaltest.mssqlspatial_ds is None:
+        pytest.skip()
+
+    source_ds = gdal.OpenEx("data/shp/testpoly.shp", gdal.OF_VECTOR)
+
+    assert source_ds
+
+    try:
+        with gdaltest.error_handler():
+            ds = gdal.VectorTranslate(
+                gdaltest.mssqlspatial_dsname,
+                "data/shp/testpoly.shp",
+                options="-nln poly_vertex_order -s_srs EPSG:32632 -t_srs EPSG:4326 -lco OVERWRITE=YES -lco GEOM_TYPE=GEOGRAPHY",
+            )
+
+            lyr = ds.GetLayerByName("poly_vertex_order")
+            assert lyr
+
+            # Simple polygon, no inner rings
+            feature = lyr.GetFeature(1)
+            geometry = feature.GetGeometryRef()
+            boundary = geometry.GetBoundary()
```

Comment:
GetBoundary() involves GEOS, and I'm not sure what it does behind the back (it could potentially modify the vertex order).
You can navigate through geometry parts with GetGeometryRef(). That is GetGeometryRef(0) on a MultiPolygon returns the first Polygon. GetGeometryRef(0) on a Polygon returns the exterior ring

Reply:
Ok, I'll change it, but Isn't the fact that this test fails without the current patch the prove that GetBoundary does not mess with vertex order at least in this test case?

Reply:
yes the current implementation of GetBoundary probably doesn't mess with those geometries. My remark was more to be immune against future GEOS behaviour changes and making it obvious we test against the raw geometry and not something that has underwent potential post processing